include/aspect/particle/property/melt_particle.h

```diff
@@ -0,0 +1,122 @@
+/*
+ Copyright (C) 2015 - 2016 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _aspect_particle_property_melt_particle_h
+#define _aspect_particle_property_melt_particle_h
+
+#include <aspect/particle/property/interface.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+      /**
+       * Implementation of a plugin in which the particle
+       * property is defined as presence of melt above the
+       * melt transport threshold. Returns a porosity of 0
+       * if melt is not present and a porosity of 1 if melt
```

Comment:
Thinking about it, I would prefer if we could avoid the term porosity in the description (here, and in the .cc file). This property is *not* the porosity, it just depends on it. Instead you could say: "This property is set to 0 if no ..., and it is set to 1 if melt is present."